### PR TITLE
Respond to HTTP POST requests with 404 in WebAssetServer

### PR DESCRIPTION
**Pull request by yjbanov**
## Description

Respond to HTTP POST requests with 404 in `WebAssetServer`. The issue seems to have been introduced with custom URL strategies, when the server started responding with HTTP 200 and serving `index.html` to all requests.

### CODE REVIEW

File: packages/flutter_tools/lib/src/isolated/devfs_web.dart [hunk 1]
```diff
@@ -333,6 +333,11 @@ class WebAssetServer implements AssetReader {
   // handle requests for JavaScript source, dart sources maps, or asset files.
   @visibleForTesting
   Future<shelf.Response> handleRequest(shelf.Request request) async {
+    if (request.method == 'POST') {
+      // Assets are served via GET only.
+      return shelf.Response.notFound('');
+    }
```
Comment (mdebbar) [inline packages/flutter_tools/lib/src/isolated/devfs_web.dart:339]: You may need to do the same in `ReleaseAssetServer.handle` but I'm not 100% sure. I'll let @jonahwilliams confirm.

File: packages/flutter_tools/lib/src/isolated/devfs_web.dart [hunk 2]
```diff
@@ -333,6 +333,11 @@ class WebAssetServer implements AssetReader {
   // handle requests for JavaScript source, dart sources maps, or asset files.
   @visibleForTesting
   Future<shelf.Response> handleRequest(shelf.Request request) async {
+    if (request.method == 'POST') {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/isolated/devfs_web.dart:336]: Should this be negated to only respond for GET instead? not that PUT is particularly common

**Review by mdebbar — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/isolated/devfs_web.dart (hunk 2 above)
Comment (yjbanov) [reply packages/flutter_tools/lib/src/isolated/devfs_web.dart:336]: Done.

File: packages/flutter_tools/lib/src/isolated/devfs_web.dart (hunk 1 above)
Comment (yjbanov) [reply packages/flutter_tools/lib/src/isolated/devfs_web.dart:339]: Done.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by yjbanov** [comment]
"Linux analyze" failure was a flake. Here's a successful rerun: https://ci.chromium.org/p/flutter/builders/try/Linux%20analyze/4074

**Review by SaavyGirl — COMMENTED** [review]
Good job

**Pull request merged**